crates/polars-io/src/cloud/options.rs

```diff
@@ -211,20 +212,29 @@ impl CloudOptions {
                     builder = builder.with_config(AmazonS3ConfigKey::Region, region.as_str())
                 },
                 None => {
-                    polars_warn!("'(default_)region' not set; polars will try to get it from bucket\n\nSet the region manually to silence this warning.");
-                    let result = reqwest::Client::builder()
-                        .build()
-                        .unwrap()
-                        .head(format!("https://{bucket}.s3.amazonaws.com"))
-                        .send()
-                        .await
-                        .map_err(to_compute_err)?;
-                    if let Some(region) = result.headers().get("x-amz-bucket-region") {
-                        let region =
-                            std::str::from_utf8(region.as_bytes()).map_err(to_compute_err)?;
-                        let mut bucket_region = BUCKET_REGION.lock().unwrap();
-                        bucket_region.insert(bucket.into(), region.into());
-                        builder = builder.with_config(AmazonS3ConfigKey::Region, region)
+                    if builder
+                        .get_config_value(&AmazonS3ConfigKey::Endpoint)
+                        .is_some()
+                    {
+                        // Set a default value if the endpoint is not aws.
+                        // See: #13042
+                        builder = builder.with_config(AmazonS3ConfigKey::Region, "us-east-1");
+                    } else {
+                        polars_warn!("'(default_)region' not set; polars will try to get it from bucket\n\nSet the region manually to silence this warning.");
```

Comment:
Btw https://docs.rs/object_store/latest/object_store/aws/fn.resolve_bucket_region.html implements this already

Reply:
Oh.. I missed that somehow.